src/content/docs/en/guides/view-transitions.mdx

```diff
@@ -642,6 +642,36 @@ At this point of the lifecycle, you could choose to define your own swap impleme
 </script>
 ```
 
+#### Building a custom swap function
+
+<p><Since v="4.15.0" /></p>
+
+The `swapFunction` object of the `astro:transitions/client` module provides five utility functions that handle specific swap-related tasks, including handling document attributes, page elements, and script execution. These functions can be used directly to define a custom swap implementation.
+
+The following example demonstrates how to use these functions to recreate Astro's built-in swap implementation:
+
+```astro
+<script>
+import { swapFunctions } from 'astro:transitions/client';
+
+// substitutes window.document with doc
+function mySwap(doc: Document) {
+  swapFunctions.deselectScripts(doc);
+  swapFunctions.swapRootAttributes(doc);
+  swapFunctions.swapHeadElements(doc);
+  const restoreFocusFunction = swapFunctions.saveFocus();
+  swapFunctions.swapBodyElement(doc.body, document.body);
+  restoreFocusFunction();
+};
+
+...
+  ev.swap = () => mySwap(ev.newDocument);
```

Comment:
```suggestion
  event.swap = () => mySwap(event.newDocument);
```
Last tiny tiny thing: assuming that `ev` stands for `event`, we generally prefer to write these words out in full as they can help people who aren't used to standard abbreviations. This would be my *preference*, if correct!

Reply:
Good one! I had used `ev` for consistency with the rest of the page. Now everything is consistent again ;)